src/averageAges.js

```diff
@@ -15,11 +15,12 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm'
+    && (!century || Math.ceil(person.died / 100) === century));
+  const totalAge = men.reduce((total, person) =>
+    total + (person.died - person.born), 0);
```

Comment:
Create a separate function for calculating average age. You use the same logic several times

Reply:
I'm sorry, don't understand, what I need to do exactly? Do I need to create a function that I can use 3 times in every task?

Reply:
Yes, as you can see you do the same in reduce, so you can create a function where you pass the same data structure and calculate average using function